### ISSUE TITLE
[New rule] prefer-define-hero-tag

### ISSUE THREAD

**Issue by lsaudon** [dart-code-checker/dart-code-metrics#1027]
**Please describe what the rule should do:**

Please describe what the rule should do:
The rule should tell us to give a `heroTag` to the widget that supports it to avoid a `FlutterError`.

**What category of rule is this? (place an "X" next to just one item)**

- [x] Warns about a potential error (problem)
- [ ] Suggests an alternate way of doing something (suggestion)
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about (it will be better if you can provide both good and bad examples):**

**❌ Bad:**
```dart
FloatingActionButton(
  onPressed: () => context.read<CounterCubit>().increment(),
  child: const Icon(Icons.add),
), // LINT
```

**✅ Good:**
```dart
FloatingActionButton(
  heroTag: 'increment_heroTag',
  onPressed: () => context.read<CounterCubit>().increment(),
  child: const Icon(Icons.add),
),
```

**Are you willing to submit a pull request to implement this rule?**

No, too complex for me.

**Comment by incendial** [comment]
@lsaudon I thought that this tag is an optional thing and the rule will lead to a lot of false-positives. Or l'm wrong?

**Comment by lsaudon** [comment]
When `heroTag` is not set, then a default `heroTag` is set.
If you have 2 widgets with default `heroTag`, when you go on the page, you have an error.
I don't know about false positives.

**Comment by incendial** [comment]
Available in 5.7.0 🚀
